src/api/java/com/minecolonies/api/colony/requestsystem/StandardRequestManager.java

```diff
@@ -0,0 +1,1275 @@
+         * @throws IllegalArgumentException when the request is already assigned
+         */
+        @SuppressWarnings("unchecked")
+        private static void assignRequest(final StandardRequestManager manager, final IRequest request) throws IllegalArgumentException
+        {
+            assignRequest(manager, request, Collections.EMPTY_LIST);
+        }
+
+        /**
+         * Method used to assign a given request to a resolver. Does take a given blacklist of resolvers into account.
+         *
+         * @param manager                The manager to modify.
+         * @param request                The request to assign.
+         * @param resolverTokenBlackList Each resolver that has its token in this blacklist will be skipped when checking for a possible resolver.
+         * @throws IllegalArgumentException is thrown when the request is unknown to this manager.
+         */
+        private static void assignRequest(final StandardRequestManager manager, final IRequest request, final IToken... resolverTokenBlackList) throws IllegalArgumentException
+        {
+            assignRequest(manager, request, Arrays.asList(resolverTokenBlackList));
+        }
+
+        /**
+         * Method used to assign a given request to a resolver. Does take a given blacklist of resolvers into account.
+         *
+         * @param manager                The manager to modify.
+         * @param request                The request to assign.
+         * @param resolverTokenBlackList Each resolver that has its token in this blacklist will be skipped when checking for a possible resolver.
+         * @throws IllegalArgumentException is thrown when the request is unknown to this manager.
+         */
+        @SuppressWarnings("unchecked")
+        private static void assignRequest(final StandardRequestManager manager, final IRequest request, final Collection<IToken> resolverTokenBlackList)
+          throws IllegalArgumentException
+        {
+            //Check if the request is registered
+            getRequest(manager, request.getToken());
+
+            Log.getLogger().debug("Starting resolver assignment search for request: " + request);
+
+            request.setState(new WrappedStaticStateRequestManager(manager), RequestState.ASSIGNING);
+
+            for (final IRequestResolver resolver : manager.requestClassResolverMap.get(request.getRequestType()))
+            {
+                //Skip when the resolver is in the blacklist.
+                if (resolverTokenBlackList.contains(resolver.getID()))
+                {
+                    continue;
+                }
+
+                //Skip if preliminary check fails
+                if (!resolver.canResolve(manager, request))
+                {
+                    continue;
+                }
+
+                @Nullable final List<IToken> attemptResult = resolver.attemptResolve(new WrappedBlacklistAssignmentRequestManager(manager, resolverTokenBlackList), request);
```

Comment:
Confused with this one =/ not sure how to do it, since `resolver.attemptResolve(new WrappedBlacklistAssignmentRequestManager(manager, resolverTokenBlackList), request);`
can be null, and a check for if it's null doesn't seem to work